File: GYM 3/GYM/nutrition-ai/app/prompts.py
def build_prompt(user_data):
    dislikes = user_data.get("dislikes", [])
    dislikes_str = ", ".join(dislikes) if dislikes else "None"

    return f"""
You are a nutritionist AI. Based on the following user details, generate a **1-day personalized meal plan** with Breakfast, Lunch, Snack, and Dinner.

Return:
1. Dish name
2. Raw ingredients with quantity (approximate weights)
3. Macronutrient & micronutrient breakdown (calories, protein, carbs, fat, fiber, iron, B12, calcium, potassium, omega-3)
4. Daily totals
5. Health notes (diabetic, allergy, weight goal, etc.)

❌ Avoid any ingredients or dishes that include: {dislikes_str}

User Profile:
- Age: {user_data['age']}
- Gender: {user_data['gender']}
- Height: {user_data['height']} cm
- Weight: {user_data['weight']} kg
- Activity Level: {user_data['activity']}
- Diet Type: {user_data['diet']}
- Cuisine Preference: {user_data['cuisine']}
- Allergies: {user_data['allergies'] or 'None'}
- Medical Conditions: {user_data['medical_conditions'] or 'None'}
- Health Goal: {user_data['goal']}

📝 Format:
Use clean Markdown with:
- `##` for each meal
- Bullet points for ingredients & nutrients
- Section at the end for totals + flags

Example:
## 🍽️ Breakfast: <Dish Name>
**🧾 Ingredients:**
- Item (weight g)
...

**🔍 Nutrition:**
- Calories: ___ kcal
- Protein: ___ g
...

## 🎯 Daily Totals
...

✅ **Flags:** diabetic-friendly, lactose-free, etc.
""".strip()

File: packages/engines/nutrition-engine/src/prompts.ts
import type { MealPlanPrompt, MealPlanRequest } from "@aura/types";

export interface CalorieQueryRequest {
  userData: MealPlanRequest;
  calorieLimit: number;
}

export interface IngredientSuggestionRequest {
  userData: MealPlanRequest;
  ingredients: string[];
}

export interface NutritionChatRequest {
  userData: MealPlanRequest;
  question: string;
}

function normalizeList(values: string[]): string {
  return values.length > 0 ? values.join(", ") : "None";
}

export function buildMealPlanPrompt(request: MealPlanRequest): MealPlanPrompt {
  return {
    userPrompt: [
      "You are a nutritionist AI. Generate a 1-day personalized meal plan with Breakfast, Lunch, Snack, and Dinner.",
      "Return dish names, raw ingredients with approximate quantities, macro and micronutrient breakdowns, daily totals, and health notes.",
      `Avoid any ingredients or dishes that include: ${normalizeList(request.dislikes)}`,
      "User Profile:",
      `- Age: ${request.age}`,
      `- Gender: ${request.gender}`,
      `- Height: ${request.heightCm} cm`,
      `- Weight: ${request.weightKg} kg`,
      `- Activity Level: ${request.activityLevel}`,
      `- Diet Type: ${request.diet}`,
      `- Cuisine Preference: ${request.cuisine}`,
      `- Allergies: ${normalizeList(request.allergies)}`,
      `- Medical Conditions: ${normalizeList(request.medicalConditions)}`,
      `- Health Goal: ${request.goal}`,
      "Format the result in Markdown with sections for each meal and a final daily totals section.",
    ].join("\n"),
  };
}

export function buildCalorieQueryPrompt(request: CalorieQueryRequest): MealPlanPrompt {
  return {
    userPrompt: [
      "You are a nutrition assistant.",
      `Suggest meals under ${request.calorieLimit} calories based on this profile:`,
      buildMealPlanPrompt(request.userData).userPrompt,
      "Return 3 to 5 meal options with calories and a short suitability note.",
    ].join("\n\n"),
  };
}

export function buildIngredientPrompt(request: IngredientSuggestionRequest): MealPlanPrompt {
  return {
    userPrompt: [
      "You are a nutrition assistant.",
      `Suggest healthy dishes using these ingredients: ${normalizeList(request.ingredients)}`,
      buildMealPlanPrompt(request.userData).userPrompt,
      "Return meal ideas with ingredient usage notes and a rough nutrition summary.",
    ].join("\n\n"),
  };
}

export function buildChatPrompt(request: NutritionChatRequest): MealPlanPrompt {
  return {
    userPrompt: [
      "You are a nutrition assistant answering a user question.",
      buildMealPlanPrompt(request.userData).userPrompt,
      `Question: ${request.question}`,
      "Respond clearly and include dietary caveats when relevant.",
    ].join("\n\n"),
  };
}
